container/go/cmd/digester_index/digester_index.go

```diff
@@ -0,0 +1,251 @@
+			"Path to the image config.json. Required if --tarball was not specified.",
+		)
+		fl.Var(
+			(*utils.ArrayStringFlags)(&imageArgs.Layers), "layer",
+			"One or more layers with the following comma separated values (Compressed layer tarball, Uncompressed layer tarball, digest file, diff ID file)."+
+				"e.g., --layer layer.tar.gz,layer.tar,<file with digest>,<file with diffID>.",
+		)
+		fl.StringVar(
+			&imageArgs.Tarball, "tarball", "",
+			"Path to the image tarball as generated by docker save. Required if --config was not specified.",
+		)
+		fl.StringVar(
+			&imageArgs.Manifest, "manifest", "",
+			"Path to the manifest of the base image. This should be the very first image in the chain of images and is only really required for windows images with a base image that has foreign layers.",
+		)
+
+		if err := fl.Parse(argv); err != nil {
+			return nil, err
+		}
+
+		if imageArgs.Tarball == "" && imageArgs.Config == "" {
+			return nil, errors.New("neither --tarball nor --config was specified")
+		}
+
+		args.Images = append(args.Images, imageArgs)
+	}
+
+	return &args, nil
+}
+
+func main() {
+	args, err := parseArgs(os.Args)
+	if err != nil {
+		if err == flag.ErrHelp {
+			os.Exit(0)
+		}
+
+		_, _ = fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(2)
+	}
+
+	if err := run(args); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run(args *applicationArgs) error {
+	var content bytes.Buffer
+
+	if len(args.Images) == 0 {
+		log.Printf("No provided images.\n")
+		return nil
+	}
+
+	ii, err := readImageIndex(args)
+	if err != nil {
+		return err
+	}
+
+	digest, err := ii.Digest()
+	if err != nil {
+		return fmt.Errorf("can't get digest of image index: %v", err)
+	}
+
+	_, _ = content.WriteString(fmt.Sprintf("%s\n", digest))
+
+	im, err := ii.IndexManifest()
+	if err != nil {
+		return fmt.Errorf("can't get index manifest: %v", err)
+	}
+
+	for _, manifest := range im.Manifests {
+		platform := manifest.Platform
+		if platform == nil {
+			platform = &v1.Platform{
+				Architecture: "amd64",
+				OS:           "linux",
+			}
+		}
+
+		var os string
+		if platform.OS != "" {
+			os = platform.OS
+		} else {
+			os = "linux"
+		}
+
+		var arch string
+		if platform.Architecture != "" {
+			arch = platform.Architecture
+		} else {
+			arch = "amd64"
+		}
+
+		platformIdent := os + "/" + arch
+		if platform.Variant != "" {
+			platformIdent += "/" + platform.Variant
+		}
+
+		_, _ = content.WriteString(fmt.Sprintf(
+			"%s\t%s\n", platformIdent, manifest.Digest,
+		))
+	}
+
+	if err := ioutil.WriteFile(args.OutputFilePath, content.Bytes(), os.ModePerm); err != nil {
+		return fmt.Errorf("error outputting digest file to %s: %v", args.OutputFilePath, err)
+	}
+
+	return nil
+}
+
+func readImageIndex(args *applicationArgs) (v1.ImageIndex, error) {
+	platforms := make([]v1.Platform, len(args.Images))
+	images := make([]v1.Image, len(args.Images))
+
+	for idx, imgArgs := range args.Images {
```

Comment:
https://google.github.io/styleguide/go/decisions#single-letter-variable-names
```suggestion
	for i, imgArgs := range args.Images {
```